micro-deps/micro-deps/src/main/java/com/ofg/infrastructure/discovery/MicroserviceConfiguration.java

```diff
@@ -20,12 +22,11 @@
 
     public MicroserviceConfiguration(ServicePath servicePath, List<Dependency> dependencies) {
         this.servicePath = servicePath;
-        this.dependencies = dependencies;
+        this.dependencies = unmodifiableList(dependencies);
```

Comment:
Those dependencies are intended to be parsed from `microservices.json` and passed to `MicroserviceConfiguration` without further modifications but using another guard against modifications (e.g. Guava immutable structures) is a good point.